virttest/qemu_devices/qcontainer.py

```diff
@@ -1018,6 +1023,27 @@ def machine_q35(cmd=False):
                 devices.append(qdevices.QStringDevice('fdc',
                                                       child_bus=qdevices.QOldFloppyBus('floppy'))
                                )
+
+            # add default pcie root port plugging pcie device
+            port_name = '%s-0' % root_port_type
+            port_params = {
+                'type': root_port_type,
+                # reserve slot 0x0 for plugging in  pci bridge
+                'reserved_slots': '0x0'}
+            root_port = self.pcic_by_params(port_name, port_params)
+            if root_port_type == 'pcie-root-port':
+                root_port.set_param('multifunction', 'on')
+                root_port.set_param('port', hex(1))
+            devices.append(root_port)
+
+            # add pci bridge for plugging in legace pci device
+            bridge_name = '%s-0' % pci_bridge_type
+            bridge_params = {'type': pci_bridge_type}
+            pci_bridge = self.pcic_by_params(bridge_name, bridge_params)
+            pci_bridge.parent_bus = ({'aobject': root_port.get_qid()},)
```

Comment:
Is there some reason about not to specify the parent bus by `bridge_params` when creating the device?

Reply:
pcic_by_params always set parent bus to pci.0, so set bridge parent bus before insert it.

Reply:
Actually, it would be set to `pci.0` only if `pci_bus` is not set in `bridge_params`.

https://github.com/avocado-framework/avocado-vt/blob/363a95460c11e86b9feafadeb4e15a409f6a83ca/virttest/qemu_devices/qcontainer.py#L2392

Reply:
set parent_bus in params works, and updated it. thanks your comments.